codestyle/druid-forbidden-apis.txt

```diff
@@ -34,6 +34,7 @@ java.util.Random#<init>() @ Use ThreadLocalRandom.current() or the constructor w
 java.lang.Math#random() @ Use ThreadLocalRandom.current()
 java.util.regex.Pattern#matches(java.lang.String,java.lang.CharSequence) @ Use String.startsWith(), endsWith(), contains(), or compile and cache a Pattern explicitly
 org.apache.commons.io.FileUtils#getTempDirectory() @ Use org.junit.rules.TemporaryFolder for tests instead
+java.lang.Class#getCanonicalName() @Class.getCanonicalName can return null for anonymous types, use Class.getName instead.
```

Comment:
on other lines I see a space between '@' and message start , it might or might not be important but we could that space just to be consistent .